src/main/anovos/data_analyzer/quality_checker.py

```diff
@@ -9,19 +10,32 @@
 from anovos.shared.utils import attributeType_segregation, transpose_dataframe, get_dtype
 from pyspark.sql import functions as F
 from pyspark.sql import types as T
+from pyspark.sql import SparkSession, DataFrame
 
 
-def duplicate_detection(spark, idf, list_of_cols='all', drop_cols=[], treatment=False, print_impact=False):
+def _parse_columns(list_of_cols: Union[List[str], str]):
```

Comment:
why are we using typing in the context

Reply:
the rest of project doesn't include any typing

Reply:
no totally agree, but will be better first just to define the types without enforcing python typing, and we can rollback second version with generalized typing:

```def duplicate_detection(spark: Sparksession, idf: Dataframe, list_of_cols: [] ='all', drop_cols: list =[], treatment: bool =False, print_impact: bool =False) -> Dataframe:
```

No idea if what i mentioned above works, need to check it